packages/eslint-plugin/.npmignore

```diff
@@ -1,6 +1,5 @@
 **/*
-!lib/**/*
+!dist/**/*
 !docs/**/*
 !package.json
-!parser.js
 !README.md
```

Comment:
we should migrate this to `files` in package.json for consistency

Reply:
the question here, is if we want to publish from the root of this package.

If we setup a post-build copy step to copy the package.json + README.md + etc into `dist`, we could then use `npm publish dist` to publish the dist folder.
This means we can flatten the folder structure, and avoid publishing a folder named `dist`.

For example, I've done this in https://github.com/bradzacher/eslint-config-brad
Note to help prevent running `npm publish` on the root folder, the base package.json includes `private: true`, which is scrubbed out during the copy into `dist`.

cc @JamesHenry

Reply:
I would also +1 the push for consistency with the other packages which all use the files array I believe

Reply:
if someone ever wants to use a file from within our packages, it's slightly uglier if you publish the package *with* the dist folder:
`require('@typescript-eslint/eslint-plugin/dist/rules/foo')`
instead of
`require('@typescript-eslint/eslint-plugin/rules/foo')`

Other than that it makes no difference!
Just a different approach!

Reply:
Why not publish the main directory (`plugin-eslint/` for example) and have files set to pick out the ones you want? Which means we can go back to having the conventional `lib` directory too and no copying files has to be done. Or am I missing something?